packages/flutter_tools/lib/src/build_system/source.dart

```diff
@@ -241,6 +259,13 @@ abstract class Source {
   /// If [artifact] points to a directory then all child files are included.
   const factory Source.hostArtifact(HostArtifact artifact) = _HostArtifactSource;
 
+  /// The source is provided by a [FlutterProject].
+  ///
+  /// If [optional] is true, the file is not required to exist. In this case, it
+  /// is never resolved as an input.
```

Comment:
Nit: Can you provide an example in the comment?

Reply:
Added!